dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -82,9 +84,95 @@ public Pattern getParameterPattern() {
         }
     }
 
+    private boolean isProcedureOrFunction(String schemaName, String objectName,
+                                                             boolean checkForFunction)
+                                                                   throws SQLException {
+        DatabaseMetaData dbmd = getConnection().getMetaData();
+        try (ResultSet rs = checkForFunction ? dbmd.getFunctions(null, schemaName, objectName) :
+                                               dbmd.getProcedures(null, schemaName, objectName)) {
+            return rs.next();
+        }
+    }
+
+    private Direction getParamDirection(int columnType, boolean isFunction) {
+        Direction dir = null;
+        if (isFunction) {
+            switch (columnType) {
+                case DatabaseMetaData.functionColumnIn:
+                    dir = Direction.INPUT;
+                    break;
```

Comment:
These could be shortcut `return` statements to shrink a bit the number of lines and to avoid the need of `dir` variable.